lib/src/simulations/epicycloid_curve.dart

```diff
@@ -0,0 +1,385 @@
+                height: 30,
+              ),
+              Slider(
+                min: 0,
+                max: 500,
+                divisions: 500,
+                activeColor: Theme.of(context).accentColor,
+                inactiveColor: Colors.grey,
+                onChanged: (animating)
+                    ? null
+                    : (value) {
+                        setState(() {
+                          total = double.parse(value.toStringAsFixed(1));
+                        });
+                      },
+                value: total,
+              ),
+              Center(
+                child: Text(
+                  (animating && animatepoints)
+                      ? "Points: Animating"
+                      : "Points: ${total.toInt()}",
+                  style: Theme.of(context).textTheme.subtitle,
+                ),
+              ),
+              Slider(
+                min: 0,
+                max: 51,
+                divisions: 510,
+                activeColor: Theme.of(context).accentColor,
+                inactiveColor: Colors.grey,
+                onChanged: (animating)
+                    ? null
+                    : (value) {
+                        setState(() {
+                          factor = double.parse(value.toStringAsFixed(1));
+                        });
+                      },
+                value: factor,
+              ),
+              Center(
+                child: Text(
+                  (animating && animatefactor)
+                      ? "Factor: Animating"
+                      : "Factor: ${factor.toStringAsFixed(1)}",
+                  style: Theme.of(context).textTheme.subtitle,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+      body: Container(
+        width: MediaQuery.of(context).size.width,
+        child: Stack(
+          children: <Widget>[
+            Epicycloid(
+              factor: factor,
+              total: total,
+              animatefactor: animatefactor,
+              animatepoints: animatepoints,
+              animating: animating,
+              key: globalKey,
+            ),
+            Positioned(
+              top: 12,
+              left: 12,
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: <Widget>[
+                  Text(
+                    'Animate with Factor -',
+                    style: TextStyle(
+                      fontSize: 16,
+                      fontWeight: FontWeight.w600,
+                    ),
+                  ),
+                  Checkbox(
+                    onChanged: (animating)
+                        ? null
+                        : (_) {
+                            setState(() {
+                              animatefactor = !animatefactor;
+                              if (animating) {
+                                animating = (animating && animatefactor);
```

Comment:
I don't get the need of this?

Reply:
I made some changes making these lines redundant so I'll remove them, thanks!